Sources/NIOPriorityQueue/PriorityQueue.swift

```diff
@@ -90,3 +86,15 @@ extension PriorityQueue: CustomStringConvertible {
         return "PriorityQueue(count: \(self.underestimatedCount)): \(Array(self))"
     }
 }
+
+extension PriorityQueue {
+  @available(*, deprecated, renamed: "Element")
+  public typealias T = Element
+  @available(*, deprecated, renamed: "Outer.Inner")
```

Comment:
Presumably `Outer.Inner` here should be explicit?

Reply:
Nope, cut+paste fail from my test to check the syntax, sorry! Should be `PriorityQueue.Iterator`.

Reply:
Not sure if it's possible to test these fixit warnings using the XCTest infrastructure (we test Swift differently)

Reply:
Fixed, sorry about that. In theory, could add a test like this to check the old names, but unfortunately this would result in the tests spitting out deprecation warning every time you run them.

```swift
    func testDeprecatedTypealiases() {
        var pq = PriorityQueue<String>(ascending: false)
        pq.push("foo")
        var it: PriorityQueue<String>.PriorityQueueIterator = pq.makeIterator()
        var t: PriorityQueue<String>.T = it.next()!
        XCTAssertEqual(t, "foo")
    }
```

Reply:
Yeah, I'm not really entirely sure XCTest supports catching or keeping track of warnings. I think that given that they're compile-time constructs the answer is probably "no".

Reply:
Yeah. [This](https://github.com/apple/swift/blob/master/test/stdlib/FlatMapDeprecation.swift#L1) is how we do it over on the std lib, but this relies on a totally different infrastructure.